py_mmd_tools/nc_to_mmd.py

```diff
@@ -38,29 +36,40 @@
 from shapely.errors import WKTReadingError
 
 
-def validate_iso8601(isodatetime):
-    regex = (
-        r'^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])'
-        r'T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-](?:2[0-3]'
-        r'|[01][0-9]):[0-5][0-9])?$'
-    )
-    match_iso8601 = re.compile(regex).match
-    valid = False
+# Converts s to a normalized ISO 8601 value: YYYY-mm-ddTHH:MM:SS<second fraction><time zone>.
+# Returns (<normalized value>, None) upon success, otherwise (None, <error reason>).
+def normalize_iso8601(s):
+    # get initial datetime
```

Comment:
We usually do Docstrings in functions. https://peps.python.org/pep-0257/. See __init__ of Nc_to_mmd class for example.

Reply:
fix coming up